### ISSUE TITLE
pop-shell on Fedora 33 Erratic window placement after Fedora update

### ISSUE THREAD

**Issue by mowestusa** [pop-os/shell#743]
**(1) Issue/Bug Description:**
After a system update of Fedora 33 on 11/23/2020, pop-shell started placing gnome terminal window erratically on the screen. Other programs seem to adhere to the stacked layout, but when a gnome terminal window is opened things become erratic impacting the placement of other windows after a terminal is opened.
- At first I thought this might be due to the fact that I had originally compiled pop-shell from source on Fedora 32, then done an upgrade to 33. However, this did not happen immediately after the full system upgrade from Fedora 32 to Fedora 33. Problems occurred after a minor system upgrade today.
- I uninstalled the pop-shell version that I compiled under Fedora 32, and installed the pop-shell extension that is now available in Fedora 33 repos. However, the same strange behavior occurred.

**(2) Steps to reproduce (if you know):**
Have pop-shell enabled, and open a gnome terminal window. I haven't been able to try every program, but gnome terminal seems to be one for sure that causes the window placement to become erratic with overlaying windows on top of one another even when stacking is turned on. Switching between floating and stacked doesn't seem to resolve the issue either. When a gnome terminal window is opened first it will take up the full screen but it will be off, shifted off of the viewable screen towards the bottom right hand corner of the screen.

**(3) Expected behavior:**
Windows would open and auto stack as normal.

**(4) Distribution (run `cat /etc/os-release`):**
 Fedora 33

**(5) Gnome Shell version:**
3.38.1 running on X11 not Wayland

**(6) Pop Shell version (run `apt policy pop-shell` or provide the latest commit if building locally):**

**(7) Where was Pop Shell installed from:**
Fedora 33 repo

**(8) Monitor Setup (2 x 1080p, 4K, Primary(Horizontal), Secondary(Vertical), etc):**
1 1920x1080

**(9) Other Installed/Enabled Extensions:**
Default ones that Fedora installs by default.

**(10) Other Notes:**

**Comment by bobbsley** [comment]
Hi!
Yesterday night I have also stumbled upon this issue. I have managed to find the culprit: for some reason the package _mutter-3.38.1-2.fc33.x86_64_ causes this odd behaviour. 
I suggest downgrading mutter to version 3.38.1-1 using _sudo dnf downgrade mutter-3.38.1-1.fc33.x86_64_ and upgrading the system with _sudo dnf upgrade -x mutter_ until someone resolves it.

**Comment by mowestusa** [comment]
This morning I noticed that an update came down for mutter-3.38.1-3.fc33.x86_64, however I can confirm that pop-shell still has the same behavior even after this upgrade of mutter, so something must still be wrong with mutter. Thanks for tracing this issue. I will keep watching for updates to mutter and trying it out. Firefox, aisleriot, and the pop-shell keyboard shortcuts windows all seemed to abide by the pop-shell arranging, but Gnome specific software is where you see the erratic behavior like with gnome terminal, files, and gedit all showed erratic behavior this morning when I re-eabled pop-shell after the mutter upgrade.

**Comment by jacobgkau** [comment]
Does the issue happen if you use the `debpatches` branch of Pop Shell from this pull request? https://github.com/pop-os/shell/pull/741

**Comment by mowestusa** [comment]
I haven't tried that, and honestly, not completely sure how I would try it. I'm guessing that I would need to compile pop-shell from a different branch than the main branch? Will this be rolled into the Fedora package for pop-shell?
I have uninstalled the compiled version that I had done earlier on Fedora 32, and I installed the pop-shell through dnf and the Fedora package instead. Sorry, not a developer, and not super savvy with git either.

**Comment by mowestusa** [comment]
Ok, I figured out how to install from the debpatches branch of Pop Shell, and it seems to work less erratically. The gnome terminal seems to work fine if I just open up 4-5 terminals on my 32" 1920x1080 monitor. If I have gnome terminal, files, and gedit open then add a 4th program (doesn't seem to matter what, the new window and the 2nd newest window overlap but don't look right. The newest window might be slightly off the screen, or the new window will be sized to tile, but the program under it will not resize so you can view both windows as the same size.
Also doing the `make local-install` from the debpatches branch doesn't seem to restore all of my pop-shell keybindings. "Super-Q" won't close a window. "Ctrl+Super+Up or Down" won't change to a different workspace. There might be others that don't work too, but I just haven't tried them.
So there seems to be some success, but not complete success. Let me know if there is something else I should try.

**Comment by Davidnet** [comment]
I also have encountered this weird issue it seems to only apply to gnome-terminal

**Comment by alaurie** [comment]
I have the same thing but only with Gnome-Terminal as @Davidnet said. I resolved this by just using another terminal app and uninstalling gnome-terminal. Sizing works perfect now everytime. I'm using elementary-terminal.

**Comment by jacobgkau** [comment]
Pop!_OS patches GNOME Terminal to resize freely (rather than its default behavior of locking to character height/width) to work better with Pop Shell. Fedora would not have that patch, so if it's just GNOME Terminal having a problem, that might be part of it.

**Comment by mowestusa** [comment]
I have seen the erratic window placement happen with Files, Gedit, and Gnome Terminal. The Gnome specific programs seem to be the issue. Other programs that are not necessarily the Gnome default apps seem to work with the intended behavior of pop-shell. Currently, I am running the debpatches branch version on Fedora 33, and I have had less issues. Sometimes if things get weird I can turn off tiling and turn it back on and windows will snap into place. Also, with the debpatches branch I don't tend to notice issues until I get above 3 programs in the same workspace. I have been just switching to a new workspace if I need to go above 3 programs on the screen. I'm still not sure why some of the pop-shell keyboard shortcuts failed to install when I installed the debpatches branch, but I fixed that by manually changing them in keyboard shortcuts.

**Comment by mmstick** [comment]
@mowestusa Is this still an issue for you with the version of Pop Shell from this repo?

**Comment by bobbsley** [comment]
@mmstick I am running the most recent version that is packaged in the Fedora repo (Commit date :20201130) and mutter-3.38.2-1.fc33.x86_64. The issue seems to be fixed now.

**Comment by mowestusa** [comment]
> @mowestusa Is this still an issue for you with the version of Pop Shell from this repo?
@mmstick,
I was able to uninstall the debpatches branch of Pop Shell that I had installed last night, and installed the version that is currently in the F33 repos. The package in the F33 repos seems to have the expected behavior and seems to function as well as the version of the extension that I had compiled from the debpatches branch. The original erratic window placement that I experienced with Gnome apps is gone. With the current package in the F33 repos I still have some windows that resize to be partially off the visible screen, but once again that only happens if I have more than 4 or 5 Gnome programs open at once. I'm guessing there is a limit to how many windows from certain tool kits can be resized. I'm not on my Fedora system at the moment, so I can't take a screen shot to show it.

At the same time, I never want more than 4 programs open on a single desktop anyway, because it makes it hard to see and use the programs when I get more than that number stacked on the screen. So for my purposes and use cases, Pop Shell is working well to organize my programs on my 32" monitor in ways that are helpful to my productivity. Thank you for fixing the original issue that I caused me to file this bug report. Thanks for sharing this wonderful extension with the wider community outside of Pop!OS.
